CMakeLists.txt

```diff
@@ -377,40 +377,53 @@ if(WITH_VTK)
   endif()
 endif()
 
-#VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND if the version we require is not found.
+#VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
 option(WITH_QT "Build QT Front-End" TRUE)
 if(WITH_QT)
-  find_package(Qt5 5.9.5 COMPONENTS Concurrent OpenGL Widgets)
+  find_package(Qt6 QUIET COMPONENTS Concurrent OpenGL Widgets)
+  if (NOT Qt6_FOUND)
+    find_package(Qt5 5.9.5 QUIET COMPONENTS Concurrent OpenGL Widgets)
+    set(Qt_FOUND ${Qt5_FOUND})
+    if(Qt5_FOUND)
+      set(QT_VERSION_MAJOR 5)
+    endif()
+  else()
+    set(Qt_FOUND TRUE)
+    set(QT_VERSION_MAJOR 6)
+  endif()
+
   set(QT_DISABLE_PRECATED_BEFORE_VAL "0x050900")
-  
-  if(Qt5_FOUND)
-    message(STATUS "Qt5 version: ${Qt5_VERSION}")
-    set(QT5_FOUND ${Qt5_FOUND})
+
+  if(Qt_FOUND)
+    message(STATUS "Qt${QT_VERSION_MAJOR} version: ${Qt${QT_VERSION_MAJOR}_VERSION}")
+
+    set(QT${QT_VERSION_MAJOR}_FOUND ${Qt_FOUND})
     #Set Cmake Auto features to skip .hh files
     if(POLICY CMP0100)
       cmake_policy(SET CMP0100 OLD)
     endif()
     
     #If building CUDA required libraries
-    #Change Qt5::Core fixed -fPIC flags to conditionally only CXX
+    #Change Qt${QT_VERSION_MAJOR}::Core fixed -fPIC flags to conditionally only CXX
     #TODO: To be removed when QT is >5.14.1
     if(BUILD_CUDA OR BUILD_GPU)
-      if(Qt5Widgets_VERSION VERSION_LESS 5.14.1)
-        get_property(core_options TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_OPTIONS)
+      if(Qt${QT_VERSION_MAJOR}Widgets_VERSION VERSION_LESS 5.14.1)
+        get_property(core_options TARGET Qt${QT_VERSION_MAJOR}::Core PROPERTY INTERFACE_COMPILE_OPTIONS)
         string(REPLACE "-fPIC" "$<IF:$<COMPILE_LANGUAGE:CXX>,-fPIC,>"  new_core_options ${core_options})
-        set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_OPTIONS ${new_core_options})
+        set_property(TARGET Qt${QT_VERSION_MAJOR}::Core PROPERTY INTERFACE_COMPILE_OPTIONS ${new_core_options})
       endif()
     endif()
     
-    get_property(core_def TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_DEFINITIONS)
+    get_property(core_def TARGET Qt${QT_VERSION_MAJOR}::Core PROPERTY INTERFACE_COMPILE_DEFINITIONS)
     list(APPEND core_def "QT_DISABLE_DEPRECATED_BEFORE=${QT_DISABLE_PRECATED_BEFORE_VAL}")
-    set_property(TARGET Qt5::Core PROPERTY INTERFACE_COMPILE_DEFINITIONS ${core_def})
+    set_property(TARGET Qt${QT_VERSION_MAJOR}::Core PROPERTY INTERFACE_COMPILE_DEFINITIONS ${core_def})
 
   else()
-    message(STATUS "Qt5 is not found.")
+    message(STATUS "Qt6/Qt5 is not found.")
   endif()
 else()
-  set(Qt5_FOUND FALSE)
+  message(STATUS "WITHOUT_QT")
```

Comment:
Was this a debug message?
I don't think we normally print things that are not selected.